app/server-routes/src/main/scala/org/bitcoins/server/routes/Server.scala

```diff
@@ -86,8 +97,49 @@ case class Server(
     httpFut.foreach { http =>
       logger.info(s"Started Bitcoin-S HTTP server at ${http.localAddress}")
     }
+    val wsFut = startWsServer()
+
+    for {
+      http <- httpFut
+      _ <- wsFut
+    } yield http
+  }
+
+  private def startWsServer(): Future[Http.ServerBinding] = {
+    val httpFut =
+      Http()
+        .newServerAt("localhost", 19999)
+        .bindFlow(wsRoutes)
+    httpFut.foreach { http =>
+      logger.info(s"Started Bitcoin-S websocket at ${http.localAddress}")
+    }
     httpFut
   }
+
+  private val tuple = {
+    //from: https://github.com/akka/akka-http/issues/3039#issuecomment-610263181
+    //the BroadcastHub.sink is needed to avoid these errors
+    // 'Websocket handler failed with Processor actor'
+    Source
```

Comment:
need to investigate a potential memory leak here if no one subscribes to the websocket and drains the queue

Reply:
This is fixed by specify the `OverflowStrategy.dropHead` which will drop the oldest element in the buffer. We have a buffer size of 25 currently, is that too little? :man_shrugging:

Reply:
This seems fine so long as its documented, but do note that another solution is to use normal backpressure but to have the application explicitly attach a trivial `Sink.ignore` to the hub so that the stream gets consumed. This is probably more idiomatic unless you explicitly want to keep old messages around for new hub subscribers for some reason

Reply:
I think the edge case to consider here is what if someone subscribed to the notifications becomes backed up and can't process the messages fast enough, do we want to drop messages or slow down the stream of messages